ironic/resource_ironic_deployment.go

```diff
@@ -94,7 +95,20 @@ func resourceDeploymentCreate(d *schema.ResourceData, meta interface{}) error {
 	// Set instance info
 	instanceInfo := d.Get("instance_info").(map[string]interface{})
 	if instanceInfo != nil {
-		_, err := UpdateNode(client, d.Get("node_uuid").(string), nodes.UpdateOpts{
+		instanceInfoCapabilities, found := instanceInfo["capabilities"]
+		capabilities := make(map[string]string)
+		nodeUUID := d.Get("node_uuid").(string)
+		if found {
+			for _, e := range strings.Split(instanceInfoCapabilities.(string), ",") {
+				parts := strings.Split(e, ":")
+				if len(parts) == 2 {
+					capabilities[parts[0]] = parts[1]
+				}
+				return fmt.Errorf("error while parsing capabilitie %s, the correct format is key:value", e)
```

Comment:
I think this returns unconditionally, and typo in capabilitie

```suggestion
else {
	return fmt.Errorf("error while parsing capability %s, the correct format is key:value", e)
}
```

Reply:
You are right, I've switched to return the error in the if in case the len is different from 2.